builtin/receive-pack.c

```diff
@@ -1118,7 +1121,7 @@ static const char *update(struct command *cmd, struct shallow_info *si)
 	}
 
 	if (do_update_worktree) {
-		ret = update_worktree(new_oid->hash);
+		ret = update_worktree(new_oid->hash, find_shared_symref("HEAD", name));
```

Comment:
can't we pass `worktree` here, instead of `find_shared_symref("HEAD", name)`.
As we had already find the worktree already at the start of the `update()`